python/langsmith/client.py

```diff
@@ -810,6 +861,14 @@ def create_run(
             "run_type": run_type,
             "execution_order": execution_order if execution_order is not None else 1,
         }
+        if not self._filter_for_sampling([run_create]):
+            return
+        if (
+            self.tracing_queue is not None
+            and run_create.get("trace_id") is not None
```

Comment:
Can we document why we have these latter conditions?

Reply:
because the backend requires calls to batch endpoint to have these 2 attrs

Reply:
ya - it's not suuper obvious without having the BE context though

Reply:
Added comments and exceptions thrown if missing